### PR TITLE
Added rot90 frontend function and test for paddle tensor manipulation

### PR DESCRIPTION
**Pull request by Shourya742**
Issue #21630 

* Added frontend function for rot90 for paddle 
* Added frontend test for rot90 for paddle 

Test Summary: 
`pytest ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py::test_paddle_rot90`

<details close>
<summary>TEST RESULTS</summary>
<br>
Screenshot 2023-08-11 at 11 47 48 AM
</details>

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/backends/paddle/experimental/manipulation.py [hunk 1]
```diff
@@ -162,6 +162,12 @@ def hstack(
             return ivy.concat(arrays, axis=0)
 
 
+with_supported_device_and_dtypes(
+    {"2.5.1 and below": {"cpu": ("bool", "int32", "int64", "float32", "float64")}},
```
Comment (sagewhocodes) [inline ivy/functional/backends/paddle/experimental/manipulation.py:166]: > x (Tensor) – The input Tensor(or LoDTensor). The data type of the input Tensor x should be float16, float32, float64, int32, int64, bool. float16 is only supported on gpu.

Dtypes `bool` and `float16` are also supported on CPU and GPU respectively, please make the required changes in the decorator.

File: ivy/functional/frontends/paddle/tensor/manipulation.py [hunk 2]
```diff
@@ -148,3 +148,20 @@ def roll(x, shifts, axis=None, name=None):
 @to_ivy_arrays_and_back
 def take_along_axis(arr, indices, axis):
     return ivy.take_along_axis(arr, indices, axis)
+
+
+@with_supported_dtypes(
+    {
+        "2.5.1 and below": (
```
Comment (sagewhocodes) [inline ivy/functional/frontends/paddle/tensor/manipulation.py:155]: > x (Tensor) – The input Tensor(or LoDTensor). The data type of the input Tensor x should be float16, float32, float64, int32, int64, bool. float16 is only supported on gpu.

Dtypes `bool` and `float16` are also supported on CPU and GPU respectively, please make the required changes in the decorator.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 3]
```diff
@@ -637,3 +640,38 @@ def test_paddle_take_along_axis(
         indices=indices,
         axis=axis,
     )
+
+
+# rot90
+@handle_frontend_test(
+    fn_tree="paddle.rot90",
+    dtype_m_k_axes=_get_dtype_values_k_axes_for_rot90(
+        available_dtypes=helpers.get_dtypes("numeric"),
```
Comment (sagewhocodes) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:649]: Make sure the updated dtypes are included in thte tests too

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 4]
```diff
@@ -637,3 +640,38 @@ def test_paddle_take_along_axis(
...
+        min_num_dims=1,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=10,
+    ),
+    test_with_out=st.just(False),
```
Comment (sagewhocodes) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:655]: What is the reason of explicitly testing with `test_with_out=False`

**Review by sagewhocodes — CHANGES_REQUESTED** [review]

**Comment by sagewhocodes** [comment]
Hello @Shourya742 , Thank you for contributing to ivy. There have been some issues with the range of dtypes that were originally supported by the backend and frontend implementations of `rot90` , could you please fix those and make sure the tests pass? 
Thanks!

File: ivy/functional/backends/paddle/experimental/manipulation.py (hunk 1 above)
Comment (Shourya742) [reply ivy/functional/backends/paddle/experimental/manipulation.py:166]: I have added `bool` to CPU already and I will add `float16` to GPU. Thank you for your response

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py (hunk 3 above)
Comment (Shourya742) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:649]: Yup will do!!

File: ivy/functional/backends/paddle/experimental/manipulation.py (hunk 1 above)
Comment (Shourya742) [reply ivy/functional/backends/paddle/experimental/manipulation.py]: @sagewhocodes I have pushed the asked changes kindly review.

**Comment by sagewhocodes** [comment]
LGTM!, Great work @Shourya742

**Pull request merged**